sdk/core/azure-core/inc/response.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <http/http.hpp>
+
+namespace Azure { namespace Core {
+
+  template <class T> class Response {
+    T m_value;
+    std::unique_ptr<Http::RawResponse> m_rawResponse;
+
+  public:
+    // Require a raw response to create a Response T
+    Response(const T& initialValue, std::unique_ptr<Http::RawResponse> rawResponse)
+        : m_value(initialValue), m_rawResponse(std::move(rawResponse))
+    {
+    }
+
+    ~Response() { this->m_value.~T(); }
+
+    // Do not give up raw response ownership.
+    Http::RawResponse& GetRawResponse() { return *this->m_rawResponse; }
+
+    T& operator=(const Response& other) = delete;
+
+    const T* operator->() const { return &this->m_value; };
+    T* operator->() { return &this->m_value; };
+    T& operator*() { return this->m_value; };
+    const T& operator*() const { return this->m_value; };
```

Comment:
I'm not personally a fan of using `*` or `->` when you aren't modeling true pointer semantics but not the end of the world.

Reply:
I agree, I really didn't see the necessity of overloading operator `*` and `->` here.

Reply:
This was my suggestion as an extremely light-weight mechanism to provide access to the payload since handling the payload is this class's primary responsibility: every use will access the payload, but not every use will care about the raw response. Typing `x->` repeatedly is much less syntactically distracting than e.g. `x.StoredValue().`.

Reply:
@CaseyCarter True, but I have never had a systematic system problem caused by needing to type loud names, but have had many such problems from unusual operator overloads. :)

If it's really such an issue that suggests that we shouldn't be modeling `Response<T>` as a wrapper for T in the first place and might want to return an aggregate-like something like `pair<T, unique_ptr<RawResponse>>` (or a named struct with similar 'just a pair of member variables' effects) instead.